javascript-bank/bank.js

```diff
@@ -0,0 +1,40 @@
+/* exported Bank */
+
+function Bank() {
+  this.nextAccountNumber = 1;
+  this.accounts = [];
+}
+
+Bank.prototype.openAccount = function (holder, balance) {
+  if (Number.isInteger(balance) && balance > 0) {
+    var newAccount = new Account(this.nextAccountNumber, holder);
+    newAccount.deposit(balance);
+    this.accounts.push(newAccount);
+    this.nextAccountNumber++;
+    return newAccount.number;
+  }
+  return null;
+};
+
+Bank.prototype.getAccount = function (number) {
+  for (var i = 0; i < this.accounts.length; i++) {
+    if (this.accounts[i].number === number) {
+      return this.accounts[i];
+    }
+  }
+  return null;
+};
+
+Bank.prototype.getTotalAssets = function () {
+  var grandTotal = 0;
+  if (!this.accounts.length) {
+    return 0;
+  }
+
+  for (var y = 0; y < this.accounts.length; y++) {
+    for (var z = 0; z < this.accounts[y].transactions.length; z++) {
+      this.accounts[y].transactions[z].type === 'deposit' ? grandTotal += this.accounts[y].transactions[z].amount : grandTotal -= this.accounts[y].transactions[z].amount;
+    }
+  }
+  return grandTotal;
+};
```

Comment:
Try using the `Account` methods instead of directly accessing their internal properties

Reply:
<img width="482" alt="Screen Shot 2022-01-28 at 12 32 05 PM" src="https://user-images.githubusercontent.com/44829206/151617315-21606611-a239-4b1a-8ec6-6ecd270c4028.png">